csrc/executor.cpp

```diff
@@ -1513,6 +1508,13 @@ std::vector<at::Tensor> FusionExecutor::runFusion(
           nullptr));
     } else {
       FUSER_PERF_SCOPE("ExecutorRunFusion::cuLaunchCooperativeKernel");
+
+      // Validation must happen after setting the attribute
+      // CU_FUNC_ATTRIBUTE_MAX_DYNAMIC_SHARED_SIZE_BYTES, otherwise we can get
+      // failures.
+      validateCooperativeLaunch(
```

Comment:
Hmm, I wonder if this is helpful if needs to be done here. The user would just always see some error, either from us or the CUDA runtime. If there's no error, the check is effectively done twice redundantly. If this is done before, and only once for each compiled kernel, it was useful as we could report the error early and the check is just one time cost.

Is it possible to do the check with the maximum shmem size when necessary?

Reply:
@naoyam I updated the PR (and description) to fail earlier - and do less checks

Reply:
I am find dropping the PR though